docker/rucio_client/scripts/updateDDMQuota

```diff
@@ -90,52 +73,93 @@ def calculate_ddm_quotas():
+        if weight == 0:
+            dm_weight = weight
+        else:
+            # This normalization sets the lowest dm_weight to 0
+            dm_weight = int(((weight - min_positive_weight) / (max(weights) - min_positive_weight)) * MAX_DM_WEIGHT)
+            #dm_weight = int((weight/sum(weights)) * MAX_DM_WEIGHT)
+        dm_weights[rse] = dm_weight
+        stats[rse]["dm_weight"] = dm_weight
 
+    return dm_weights
 
-def set_ddm_quotas():
-    # Set automatically calculated ddm quotas
-    for rse, ddm_quota in ddm_quotas.items():
-        if DRY_RUN:
-            print("DRY-RUN: Set ddm_quota for {} to {}".format(rse, ddm_quota))
-        else:
-            client.add_rse_attribute(rse, "ddm_quota", ddm_quota)
-            print("Set ddm_quota for {} to {}".format(rse, ddm_quota))
 
-    # Set overriden ddm_quotas
-    for rse, ddm_quota in overridden_ddm_quotas.items():
+def set_dm_weights(dm_weights):
+    # Set automatically calculated dm weights
+    for rse, dm_weight in dm_weights.items():
         if DRY_RUN:
-            print("DRY-RUN: Override ddm_quota for {} to {}".format(rse, ddm_quota))
+            print("DRY-RUN: Set dm_weight for {} to {}".format(rse, dm_weight))
         else:
-            client.add_rse_attribute(rse, "ddm_quota", ddm_quota)
-            print("Override ddm_quota for {} to {}".format(rse, ddm_quota))
-
+            client.add_rse_attribute(rse, "dm_weight", dm_weight)
+            # Keep setting ddm_quota for disk rses until we deprecate it
+            if "Tape" not in rse:
+                client.add_rse_attribute(rse, "ddm_quota", dm_weight)
+            print("Set dm_weight for {} to {}".format(rse, dm_weight))
+
+def run(rse_expression, static_weight, free_weight, expired_weight, make_quadratic):
+
+    # Calculate dm_weights for disk rses
+    # T3s do not have the "static" usage set, they are quasi-static
+    dm_weights = calculate_dm_weights(rse_expression = rse_expression,
+                         static_weight = static_weight,
+                         free_weight = free_weight, 
+                         expired_weight = expired_weight,
+                         make_quadratic = make_quadratic)
+    normalized_dm_weights = normalize_dm_weights(dm_weights)
+    set_dm_weights(normalized_dm_weights)
 
 def main():
 
-    calculate_ddm_quotas()
-    normalize_ddm_quotas()
-    set_ddm_quotas()
-    if VERBOSE:
-        get_stats()
+    # Calculate dm_weights for disk rses
+    # T3s do not have the "static" usage set, they are quasi-static
+    
+    run(rse_expression = "rse_type=DISK&cms_type=real&tier<3&tier>0",
+                         static_weight = 0.2,
+                         free_weight = 0.5, 
+                         expired_weight = 0.3,
+                         make_quadratic = True)
+    
+
+    # Calculate dm_weights for tape rses
+    run(rse_expression = "rse_type=TAPE&wmcore_output_tape=True\cms_type=test",
```

Comment:
Which tapes do not have `wmcore_output_tape=True`? 
If we already have `wmcore_output_tape=True` as an attribute that defines if data goes or not on the tape. Then we do not need to explicitly set `dm_weight` (or `ddm_quota`) to `0`. 

Also, do we not have `cms_type=real` for the TAPE rses?
We shall prefer it for consistency with the disk expression.

Reply:
CERN, JINR and MIT Tape aren't used for prod output and we're not setting `dm_weight` for them. However, @amaltaro was asking to set the attribute for them as well: [link to jira](https://its.cern.ch/jira/browse/CMSTRANSF-853?focusedId=6342076&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-6342076), but it's not clear whether this is strongly necessary, so I keep those RSEs untouched for now.

> Also, do we not have cms_type=real for the TAPE rses?
> We shall prefer it for consistency with the disk expression.

We do, I'll update it.